src/actions/tamanu/prepare_upgrade.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::{bail, IntoDiagnostic, Result, WrapErr};
+use node_semver::Version;
+use regex::Regex;
+use tracing::info;
+
+use crate::actions::{
+	tamanu::{
+		download::{download, make_url},
+		find_package,
+	},
+	Context,
+};
+
+use super::{download::ServerKind, find_existing_version, find_tamanu, TamanuArgs};
+
+/// Setup new Tamanu instllation without taking down the existing one
+#[derive(Debug, Clone, Parser)]
+pub struct PrepareUpgrade {
+	/// Version to update to.
+	#[arg(value_name = "VERSION")]
+	pub version: Version,
+
+	/// Package to upgrade.
+	///
+	/// By default, this command looks for the most recent installed version of Tamanu.
+	/// If both central and facility servers are present and
+	/// configured, it will pick one arbitrarily.
+	#[arg(short, long)]
+	pub package: Option<String>,
+
+	/// Force installing older Tamanu
+	#[arg(long)]
+	pub force: bool,
```

Comment:
```suggestion
	pub force_downgrade: bool,
```

That will make it clearer and we may want to have other kinds of "forcing" behaviour down the line.